src/core/exchange/exchange-selectors.js

```diff
@@ -102,6 +118,11 @@ export function getExchangeRate(
 
   // Only search if the endpoints exist:
   if (search.routes[fromCurrency] && search.routes[toCurrency]) {
+    // Make sure that routes eventually point to a fiat currency
+    if (!hasDeepFiatRoute(search, fromCurrency))
```

Comment:
I feel like the previous version had another `!isFiatCode(fromCurrency)`, which is now missing. Maybe that check belongs in the `hasDeepFiatRoute` method as a quick pre-check before doing the loop?